tutorial/marine/quick_start.md

```diff
@@ -0,0 +1,297 @@
+🔧   Creating project called `greeter`...
+✨   Done! New project created /workspaces/devcontainer/greeter
+root@56892f4726bb:/workspaces/devcontainer#
+```
+
+We initiated a module project call `greeter` which is now visible in the VSCode Explorer with the following directories and files:
+
+* scripts
+* src
+* Cargo.toml
+* Config.toml
+
+Let's unpack.
+
+The Cargo.tom file, see below, is pretty standard Rust fare except for:
+
+1. We need to specify the binary information for our wasm32-wasi compile target
+2. We add the [Fluence](https://github.com/fluencelabs/fluence) crate to the dependencies, which includes the marine-rs sdk and the Marine-specific Wasm macros
+3. We add the fluence-test crate to the dev dependencies which allows us to write "regular" Rust unit-tests that can be run with `cargo test`
+
+```toml
+# Cargo.toml
+[package]
+name = "greeter"
+version = "0.1.0"
+authors = ["boneyard93501 <4523011+boneyard93501@users.noreply.github.com>"]
+edition = "2018"
+description = "greeter, a Marine wasi module"
+license = "Apache-2.0"
+
+[[bin]]    <-- 1
+name = "greeter"
+path = "src/main.rs"
+
+[dependencies] <-- 2
+fluence = { version="0.6.9", features=["logger"]}
+log = "0.4.14"
+
+[dev-dependencies]  <-- 3
+fluence-test = "0.1.9"
+
+[dev]
+[profile.release]
+opt-level = "s"
+```
+
+The Config.toml file provides the meta data necessary to properly instrument a Wasm IT module for execution by the Marine runtime.
+
+```toml
+# Config.toml
+modules_dir = "artifacts/"   <-- 1
+
+[[module]]  <-- 2
+    name = "greeter"
+    mem_pages_count = 1
+    logger_enabled = true
+```
+
+As illustrated in Figure 1 above, the Config.toml file is required for the local execution of Wasm modules such as the Repl or unittests. A Config.toml files consists of:
+
+1. Location of the wasm module to be executed by Marine
+2. One or mode module specific runtime requirements such as module name, logging particulars, local access and memory allocation, Recall, [Wasm memory](https://webassembly.github.io/spec/core/exec/runtime.html#memory-instances) is allocated in linear pages of 64Ki each.
+
+As evident by the existence of the main.rs file, Marine Wasm modules are created as binaries and as we see below, the Marine toolchain compiles our Rust code to the desired wasm32-wasi target. Due to the Wasm Interface Types (IT) target, our Rust files look a little different than Rust application or even wasm-bindgen code.
+
+```rust
+// main.rs
+use fluence::{marine, module_manifest, WasmLoggerBuilder};  // <-- 1
+
+module_manifest!();  <-- 2
+
+pub fn main() { // <-- 3
+    WasmLoggerBuilder::new().build().ok();  // <-- 4
+}
+
+#[marine]  // <-- 5
+pub fn greeting(name: String) -> String {  // <-- 6
+    format!("Hi, {}", name)
+}
+
+#[cfg(test)]
+mod tests {
+    use fluence_test::marine_test; // <-- 7
+
+    #[marine_test(config_path = "../Config.toml", modules_dir = "../artifacts")]
+    fn test_greeting() {
+        let name = "Marine".to_string();
```

Comment:
Suggest instead of making a `String` here (that requires allocation), it's possible to define just `&str` and then change `name.clone()` to `name.to_string()`. With that change it's possible to avoid one excess allocation.